Framework/Algorithms/src/PDCalibration.cpp

```diff
@@ -1167,12 +1167,8 @@ std::set<detid_t> PDCalibration::detIdsForTable() {
     return detids;
 
   // get the indices to loop over
-  std::size_t startIndex = 0;
-  if (!isDefault("StartWorkspaceIndex"))
-    startIndex = static_cast<std::size_t>(m_startWorkspaceIndex);
-  std::size_t stopIndex = m_uncalibratedWS->getNumberHistograms();
-  if (!isDefault("StopWorkspaceIndex"))
-    stopIndex = static_cast<std::size_t>(m_stopWorkspaceIndex);
+  std::size_t startIndex = static_cast<std::size_t>(m_startWorkspaceIndex);
+  std::size_t stopIndex = static_cast<std::size_t>(m_stopWorkspaceIndex);
```

Comment:
I think you need to put back the `!isDefault` otherwise this forces the user to specify the optional `StopWorkspaceIndex` parameter. The default value for `StartWorkspaceIndex` is already zero so that one is ok to drop.

Reply:
When `m_starWorkspaceIndex` is initialixed in L466, it is set to either the user input value, or to the last index in the input workspace.  So by the time it is called here, it will have a valid value.  I can add it anyway for extra safety

Reply:
`m_stopWorkspaceIndex` is set correctly there too